### PR TITLE
chore: wait for specs page to load before continuing in tests in order to reduce flake

### PR DESCRIPTION
**Pull request by jennifer-shehane**
## Problem

I've noticed A LOT of flaky failures around our main navigation within the launchpad (like, clicking on the lefthand sidebar and the new page not being loaded). 

One thing I noticed while reviewing one of those flaky cases was that we visit the App route (by default this is the specs list) and then immediately click to another nav. 

In the flaky test case, the Specs list was not done fully loading as shown below. See [Test Replay.](https://cloud.cypress.io/projects/ypt4pf/analytics/flaky-tests?att=2&branches=%5B%7B%22value%22%3A%22develop%22%2C%22label%22%3A%22develop%22%7D%5D&browsers=%5B%5D&chartRangeMostCommonErrors=%5B%5D&chartRangeSlowestTests=%5B%5D&chartRangeTopFailures=%5B%5D&committers=%5B%5D&cypressVersions=%5B%5D&flaky=%5B%5D&operatingSystems=%5B%5D&replayTestId=2de02f94-08cf-4f94-b04e-37852f02e2ba&runGroups=%5B%5D&specFiles=%5B%5D&status=%5B%7B%22label%22%3A%22Passed%22%2C%22value%22%3A%22PASSED%22%7D%2C%7B%22label%22%3A%22Failed%22%2C%22value%22%3A%22FAILED%22%7D%5D&tags=%5B%5D&tagsMatch=ANY&timeInterval=WEEK&timeRange=%7B%22startDate%22%3A%222024-01-26%22%2C%22endDate%22%3A%222024-02-02%22%7D&ts=1706894330253.3994&viewBy=TEST_CASE)

Screenshot 2024-02-02 at 1 28 38 PM

In the passing test case, the Specs list was done fully loading before clicking to the next nav. See [Test Replay.](https://cloud.cypress.io/projects/ypt4pf/analytics/flaky-tests?att=3&branches=%5B%7B%22value%22%3A%22develop%22%2C%22label%22%3A%22develop%22%7D%5D&browsers=%5B%5D&chartRangeMostCommonErrors=%5B%5D&chartRangeSlowestTests=%5B%5D&chartRangeTopFailures=%5B%5D&committers=%5B%5D&cypressVersions=%5B%5D&flaky=%5B%5D&operatingSystems=%5B%5D&replayTestId=2de02f94-08cf-4f94-b04e-37852f02e2ba&runGroups=%5B%5D&specFiles=%5B%5D&status=%5B%7B%22label%22%3A%22Passed%22%2C%22value%22%3A%22PASSED%22%7D%2C%7B%22label%22%3A%22Failed%22%2C%22value%22%3A%22FAILED%22%7D%5D&tags=%5B%5D&tagsMatch=ANY&timeInterval=WEEK&timeRange=%7B%22startDate%22%3A%222024-01-26%22%2C%22endDate%22%3A%222024-02-02%22%7D&ts=1706894338527&viewBy=TEST_CASE)

Screenshot 2024-02-02 at 1 28 58 PM

## Proposed Solution

For this update, I've added a check in a custom command to ensure the specs list is shown before moving on to the next step. Hoping this will prevent the flake we've been seeing.

### Edge cases

Adding this check did add a couple of other cases we need to consider. I added some logic to account for these. Sometimes the default page is for a new project, so it shows this create specs page 

Screenshot 2024-02-05 at 2 13 08 PM

And sometimes it's for a project with a custom specPattern where we show this 'no specs found' page. 

Screenshot 2024-02-05 at 2 12 54 PM

## Amendment

After changing the logic to wait for the specs page to be fully loaded, I noticed a spec that was then consistently failing. We were not actually testing what we should have been in this test!. Check out the [passing Test Replay](https://cloud.cypress.io/projects/ypt4pf/runs/53867/test-results/18f91207-e913-4cc5-b6e2-68b15e15c523/replay?actions=%5B%5D&att=1&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&pc=log-http%3A%2F%2Flocalhost%3A4455-630__command-logs&specs=%5B%7B%22label%22%3A%22top-nav.cy.ts%22%2C%22value%22%3A%22%5B%5C%2291b7b6fa-02be-4317-8826-e6b1b9926547%5C%22%5D%22%7D%5D&statuses=%5B%5D&testingTypesEnum=%5B%5D&ts=1707156179898.2) and the newly [failing Test Replay](https://cloud.cypress.io/projects/ypt4pf/runs/53870/overview/623622c6-f1c9-4c3c-a378-7bbc61568892/replay?att=1&pc=log-http%3A%2F%2Flocalhost%3A4455-673__command-logs&roarHideRunsWithDiffGroupsAndTags=1&ts=1707162736073.4). I updated the timing of when the profile request sends unauthorized to properly test what we wanted. I also updated the 'loginButton' to be a 'profileButton' since this reflect what it actually is - this was really confusing to me to begin with. 

Screenshot 2024-02-05 at 3 02 09 PM
Screenshot 2024-02-05 at 3 04 28 PM

### Steps to test

I ran the debug test locally and the assertions passed by themselves. Ensure the other tests pass and monitor for flake.

### How has the user experience changed?

N/A

### PR Tasks
- N/A Have tests been added/updated?
- N/A Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- N/A Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
Have to rethink this because sometimes it's the empty specs page that shows, which doesn't have the spec-list-container. 
Screenshot 2024-02-02 at 2 14 21 PM

File: packages/frontend-shared/cypress/support/e2e.ts [hunk 1]
```diff
@@ -360,7 +360,26 @@ function visitApp (href?: string, opts?: Partial<Cypress.VisitOptions>) {
...
+      // just make sure the page is done rendering before taking any action
+      if (appSetup === 'new-project') {
+        // if this is a new project, we'll be on the create spec page
+        return cy.get('[data-cy=create-spec-page-cards]').should('be.visible')
+      }
+
+      if (appSetup === 'no-specs') {
+        // if this is an existing project with no specs, we'll be on the no specs found page
+        return cy.get('[data-cy=create-spec-page-description]').should('be.visible')
+      }
```
Comment (emilyrohrbough) [inline packages/frontend-shared/cypress/support/e2e.ts]: Is the `spec-list-container` rendered with these project setups as well?

Comment (jennifer-shehane) [reply packages/frontend-shared/cypress/support/e2e.ts]: Sadly no. I'm leaning towards creating a new custom command to just verify the specs page is present, so going to try that to not overload the visitApp command

Comment (emilyrohrbough) [reply packages/frontend-shared/cypress/support/e2e.ts]: I'm surprised our UI changes these values so much

**Review by emilyrohrbough — APPROVED** [review]

**Review by cacieprins — APPROVED** [review]

**Pull request merged**